Unit tests: Refactor validator tests

MathyFurret opened the pull request:
Team validator unit tests are now split into several files under a `team-validator` directory:
* `basic.js` - Basic functionality
* `breeding.js` - Related to breeding and egg moves
* `custom-rules.js` - All the tests from the "custom rules" section
* `events.js` - Tests related to event Pokemon
* `formes.js` - Tests related to Pokemon with multiple formes
* `misc.js` - Tests too specific to be considered "basic" but that didn't fit anywhere else

Some stupidly long tests with extremely broad assertions were broken down.

Let me know if there's any tests you would move to a different file.

DaWoblefet commented on test/sim/team-validator/basic.js:
```diff
@@ -0,0 +1,302 @@
+		assert.equal(illegal, null);
+	});
+
+	it('should validate the Diancie released with zero perfect IVs', function () {
```
events.js

DaWoblefet commented:
Everything else looks good!

One of the things that's always bothered me is how we use ``assert(illegal)`` or ``assert.equals(illegal, null)``, which I think isn't very intuitive and has tripped me up more than once (e.g. typoing ``assert(illegal, null)``). What do you think about also refactoring to use a function like ``assert.legalTeam`` and explicitly the result of ``ValidateTeam`` along with ``true`` or ``false``? This might be a prospect for down the road and not in this PR.

MathyFurret commented:
Loved the idea so much that I implemented it right away. Wasn't too hard with a regex replace. I also moved that 1 test as you requested.

I think some other tests need to be refactored to use `assert.false.method` as well

DaWoblefet commented:
Really good work, thanks Mathy!
